drivers/spi/spi_xec_gpspi.c

```diff
@@ -0,0 +1,791 @@
+		return -ENOTSUP;
+	}
+
+	if (SPI_WORD_SIZE_GET(spi_conf->operation) != 8) {
+		LOG_ERR("Word size != 8 not supported");
+		return -ENOTSUP;
+	}
+
+	return 0;
+}
+
+static int req_full_reconfig(const struct device *dev, const struct spi_config *spi_conf)
+{
+	struct spi_xec_gpspi_data * const data = dev->data;
+
+	if (data->scfg.frequency != spi_conf->frequency) {
+		return 1; /* frequency change */
+	} else if ((data->scfg.operation & SPI_MODE_MASK)
+		   != (spi_conf->operation & SPI_MODE_MASK)) {
+		return 1; /* CPOL and/or CPHA changed */
+	}
+
+	return 0;
+}
+
+/* Configure GPSPI for full-duplex operation.
+ * TODO - configuration will become complex if driver supports CONFIG_SPI_EXTENDED_MODES
+ * If target device is a SPI flash the commands for dual/quad involve multiple phases:
+ * transmit command, transmit address, optional transmit mode byte, optional transmit
+ * clocks with I/O tri-stated, read data using multiple I/O pins.
+ * The SPI driver specifies the I/O mode in struct spi_config on each call to the
+ * transfer API. This means struct spi_config contents can change on every call and
+ * the previous call may have left the SPI open (chip select asserted). For this case
+ * reconfiguring the controller should not cause any SPI signals to glitch.
+ * Rules:
+ * If GPSPI.control.CE is 1 then chip select is asserted
+ *   Ignore frequency change. Should we return error if new frequency?
+ *   Record other flags such as SPI_HOLD_ON_CS and SPI_LOCK_ON
+ *   Should we allow MSBF/LSBF change?
+ *   Allow I/O mode change: half-duplex, full-duplex, and dual. Reject quad.
+ * Else
+ *   Allowed to do full reconfigure (controller reset)
+ * Endif
+ *
+ */
+static int gpspi_configure(const struct device *dev, const struct spi_config *spi_conf)
+{
+	const struct spi_xec_gpspi_config * const cfg = dev->config;
+	struct spi_xec_gpspi_data * const data = dev->data;
+	struct xec_gpspi_regs * const regs = cfg->regs;
+	struct spi_context *ctx = &data->ctx;
+	int ret;
+	uint8_t ctrl;
+
+	/* Need to reconfigure? */
+	ret = memcmp(&data->scfg, spi_conf, sizeof(struct spi_config));
+	if (ret == 0) {
+		return 0;
+	}
+
+	ret = gpspi_check_unsupported_features(spi_conf);
+	if (ret) {
+		return ret;
+	}
+
+	if (req_full_reconfig(dev, spi_conf)) {
+		if (!(regs->control & BIT(XEC_GPSPI_CTRL_CE_POS))) {
+			xec_gpspi_reset(regs);
+		}
+		gpspi_configure_spi_clock(regs, spi_conf->frequency);
+		gpspi_set_spi_mode(dev, spi_conf);
+	}
+
+	ctrl = regs->control & ~(XEC_GPSPI_CTRL_SPDIN_SEL_MSK | BIT(XEC_GPSPI_CTRL_LSBF_POS));
+	ctrl |= BIT(XEC_GPSPI_CTRL_BI_DIR_OUT_EN_POS);
+
+	if ((spi_conf->operation & SPI_LINES_MASK) == SPI_LINES_DUAL) {
+		ctrl |= XEC_GPSPI_CTRL_SPDIN_SEL_DUAL;
+	} else {
+		ctrl |= XEC_GPSPI_CTRL_SPDIN_SEL_FULL_DUPLEX;
+	}
+
+	if (spi_conf->operation & SPI_TRANSFER_LSB) {
+		ctrl |= BIT(XEC_GPSPI_CTRL_LSBF_POS);
+	}
+	regs->control = ctrl;
+
+	ctx->config = spi_conf;
+
+	memcpy(&data->scfg, spi_conf, sizeof(struct spi_config));
```

Comment:
spi_conf is not meant to change (though spi api does not strictly make it constant), since it must come from dts.
So ctx has the pointer, no need no copy the thing again.

Reply:
Here is the first place value is getting assigned to ctx->config.  Device info will be read from application through DTS and passed as argument as spi_conf. So very first we need to assign ctx->config.